### ISSUE TITLE
Stable / tests ?

### ISSUE THREAD

**Issue by joeblew99** [chonthu/aferoS3#1]
I am wanting to use this on a project.
How stable would you guess it is ? 

I have a project where I have 100's of servers in remote locations. In each location the server serves files to  the local clients.

All of the remote servers need to refresh their files, but only when a data base row is added locally that has a field that contains a s3 path.
So I could easily, watch for s3 paths being added to the dB, and when this occurs, kick off a goroutine that tries to download it from s3.

The middler code of the server,  can also do the same pattern. If it sees a s3 path, it will. Try to get the file locally, and if not there, pull it from s3.

The whole point is that the paths are the same for local and s3.
Basically local file system is acting as a cache, with s3 as the master.

Sorry about the long use case, 

Thanks in advance
